containers-tests/tests/intmap-strictness.hs

```diff
@@ -68,12 +70,35 @@ prop_strictFromSet :: Func Key (Bot A) -> IntSet -> Property
 prop_strictFromSet fun set =
   isBottom (M.fromSet f set) === any (isBottom . f) (IntSet.toList set)
   where
-    f = coerce (applyFunc fun) :: Key -> A
+    f = applyFunc fun
+
+prop_strictFromSetA :: Func Key (Bot A) -> IntSet -> Property
+prop_strictFromSetA fun set =
+  isBottom (getSolo (M.fromSetA (MkSolo . f) set)) === any (isBottom . f) (IntSet.toList set)
+  where
+    f = applyFunc fun
```

Comment:
This is a weird way to do this.

What we want in the test here is a `Key -> Solo A` where `A` can be bottom. That translates to generating an arbitrary `Func Key (Solo (Bot A))`, which can be coerced to `Key -> Solo A`.

Reply:
`Func Key (Bot A)` and `Func Key (Solo (Bot A))` are equivalent, and the only difference is whether you unwrap the Solo in the RHS or you wrap with Solo on the LHS. I preferred the former because it's clearer what the functions is meant to do - produce a bottom value from a key.